### PR TITLE
ARCHETYPE-487 Add input validation for required properties defined in…

### PR DESCRIPTION
**Pull request by mryan43**
This pull request implements the following new feature : https://issues.apache.org/jira/browse/ARCHETYPE-487

### CODE REVIEW

**Comment by mryan43** [comment]
21 days and nobody replied... is this project dead ?

**Comment by michael-o** [comment]
There hasn't much activity lately. Ping me again next week and I'll review and merge your PR.

**Comment by mryan43** [comment]
Any news on this one ?

**Comment by LIttleAncientForestKami** [comment]
Hiya, @michael-o, can anything be done here to help?

File: maven-archetype-plugin/src/main/java/org/apache/maven/archetype/ui/ArchetypeConfiguration.java [hunk 1]
```diff
@@ -19,12 +19,15 @@
  * under the License.
  */
 
+import org.apache.maven.archetype.common.Constants;
+import org.codehaus.plexus.util.StringUtils;
```
Comment (michael-o) [inline maven-archetype-plugin/src/main/java/org/apache/maven/archetype/ui/ArchetypeConfiguration.java]: Unrelated change, please drop.

File: maven-archetype-plugin/src/main/java/org/apache/maven/archetype/ui/DefaultArchetypeFactory.java [hunk 2]
```diff
@@ -79,9 +78,8 @@ else if ( defaultValue != null )
         getLogger().debug( "Setting property " + key + "=" + configuration.getProperty( key ) );
     }
 
-    @SuppressWarnings( "checkstyle:linelength" )
-    public ArchetypeConfiguration createArchetypeConfiguration( org.apache.maven.archetype.old.descriptor.ArchetypeDescriptor archetypeDescriptor,
-                                                                Properties properties )
+    public ArchetypeConfiguration createArchetypeConfiguration(
+        org.apache.maven.archetype.old.descriptor.ArchetypeDescriptor archetypeDescriptor, Properties properties )
```
Comment (michael-o) [inline maven-archetype-plugin/src/main/java/org/apache/maven/archetype/ui/DefaultArchetypeFactory.java]: Unrelated change, please drop.

File: maven-archetype-plugin/src/main/java/org/apache/maven/archetype/ui/DefaultArchetypeFactory.java [hunk 3]
```diff
@@ -110,9 +108,8 @@ private void addRequiredProperty( ArchetypeConfiguration configuration, Properti
         }
     }
 
-    @SuppressWarnings( "checkstyle:linelength" )
-    public ArchetypeConfiguration createArchetypeConfiguration( org.apache.maven.archetype.metadata.ArchetypeDescriptor archetypeDescriptor,
-                                                                Properties properties )
+    public ArchetypeConfiguration createArchetypeConfiguration(
+        org.apache.maven.archetype.metadata.ArchetypeDescriptor archetypeDescriptor, Properties properties )
```
Comment (michael-o) [inline maven-archetype-plugin/src/main/java/org/apache/maven/archetype/ui/DefaultArchetypeFactory.java]: Unrelated change, please drop.

File: maven-archetype-plugin/src/main/java/org/apache/maven/archetype/ui/DefaultArchetypeFactory.java [hunk 4]
```diff
@@ -216,8 +220,8 @@ public ArchetypeConfiguration createArchetypeConfiguration( MavenProject project
                 configuration.addRequiredProperty( requiredProperty );
 
                 configuration.setProperty( requiredProperty, properties.getProperty( requiredProperty ) );
-                getLogger().debug( "Setting property " + requiredProperty + "="
-                                       + configuration.getProperty( requiredProperty ) );
+                getLogger().debug(
+                    "Setting property " + requiredProperty + "=" + configuration.getProperty( requiredProperty ) );
```
Comment (michael-o) [inline maven-archetype-plugin/src/main/java/org/apache/maven/archetype/ui/DefaultArchetypeFactory.java]: Unrelated change, please drop.

File: maven-archetype-plugin/src/main/java/org/apache/maven/archetype/ui/generation/DefaultArchetypeGenerationConfigurator.java [hunk 5]
```diff
@@ -197,7 +197,11 @@ else if ( archetypeArtifactManager.isOldArchetype( ad.getGroupId(), ad.getArtifa
                                 value = getTransitiveDefaultValue( value, archetypeConfiguration, requiredProperty,
                                                                    context );
 
-                                value = archetypeGenerationQueryer.getPropertyValue( requiredProperty, value );
+                                value = archetypeGenerationQueryer.getPropertyValue(
+                                    requiredProperty,
+                                    value,
+                                    archetypeConfiguration.getPropertyValidationRegex( requiredProperty )
+                                );
```
Comment (michael-o) [inline maven-archetype-plugin/src/main/java/org/apache/maven/archetype/ui/generation/DefaultArchetypeGenerationConfigurator.java]: Condense in two lines please.

**Comment by michael-o** [comment]
Just added a few comments. When they are resolved, I will merge.

**Comment by stain** [comment]
Hi, @mryan43, have you had a chance to review @michael-o's comments on this PR?

**Comment by michael-o** [comment]
@stain I did already in April and wrote how to improve the PR. Nothing has happened.

File: maven-archetype-plugin/src/main/java/org/apache/maven/archetype/ui/ArchetypeConfiguration.java (hunk 1 above)
Comment (mryan43) [inline maven-archetype-plugin/src/main/java/org/apache/maven/archetype/ui/ArchetypeConfiguration.java]: Dropping this breaks compilation...

Comment (michael-o) [inline maven-archetype-plugin/src/main/java/org/apache/maven/archetype/ui/ArchetypeConfiguration.java]: This is still an unrelated change because those imports are already present a few lines below and have been moved to this position. An unrelated change for me.

Comment (mryan43) [inline maven-archetype-plugin/src/main/java/org/apache/maven/archetype/ui/ArchetypeConfiguration.java]: okay, I'll put it back to the wrong place (it's the only file where the org.* imports are after the java.* imports)

Comment (michael-o) [inline maven-archetype-plugin/src/main/java/org/apache/maven/archetype/ui/ArchetypeConfiguration.java]: Yes, though it was the right thing to do. It is still unrelated to this PR. Code improvements are always welcome and shall be made with separate PRs.

Thank you!

**Comment by michael-o** [comment]
@mryan43 Looks way better now. Please squash, I will test and pull.

**Comment by mryan43** [comment]
squashed

**Comment by ptahchiev** [comment]
@michael-o Michael, are you happy with the squash? Let's move this forward I want to plan a release of the archetype.

**Comment by michael-o** [comment]
@ptahchiev Thanks for pinping, this got lost on my desk. I will have a look later this day.

**Comment by michael-o** [comment]
@ptahchiev Merged. Watchout, there is a test error on master, not related to this merge.

**Comment by michael-o** [comment]
@mryan43 Please close.

**Comment by mryan43** [comment]
thx

**Comment by jira-importer** [comment]
Resolve #759

**Comment by jira-importer** [comment]
Resolve #759
